src/fast_radiology/models/unet.py

```diff
@@ -0,0 +1,111 @@
+""" Parts of the U-Net model """
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class DoubleConv(nn.Module):
+    """(convolution => [BN] => ReLU) * 2"""
+
+    def __init__(
+        self, in_channels, out_channels, mid_channels=None, with_dropout=False
+    ):
+        super().__init__()
+        if not mid_channels:
+            mid_channels = out_channels
+
+        conv1 = nn.Conv2d(in_channels, mid_channels, kernel_size=3, padding=1)
+        nn.init.kaiming_normal_(conv1.weight)
+        conv2 = nn.Conv2d(mid_channels, out_channels, kernel_size=3, padding=1)
+        nn.init.kaiming_normal_(conv2.weight)
+
+        seq = [conv1, nn.ReLU(inplace=True), conv2, nn.ReLU(inplace=True)]
+        if with_dropout:
+            seq.append(nn.Dropout(p=0.5))
+
+        self.double_conv = nn.Sequential(*seq)
+
+    def forward(self, x):
+        return self.double_conv(x)
+
+
+class Down(nn.Module):
+    """Downscaling with maxpool then double conv"""
+
+    def __init__(self, in_channels, out_channels, with_dropout=False):
+        super().__init__()
+        self.maxpool_conv = nn.Sequential(
+            nn.MaxPool2d(2),
```

Comment:
Initially it looked suspicious, but after review against the original image, putting MaxPool before convolutions here seems to be convenient code solution. In the original image skip-connections happen before MaxPool, which is what will happen here.